packages/flutter/lib/src/rendering/sliver_padding.dart

```diff
@@ -346,6 +297,82 @@ class RenderSliverPadding extends RenderSliver with RenderObjectWithChildMixin<R
       return true;
     }());
   }
+}
+
+/// Insets a [RenderSliver], applying padding on each side.
+///
+/// A [RenderSliverPadding] object wraps the [SliverGeometry.layoutExtent] of
+/// its child. Any incoming [SliverConstraints.overlap] is ignored and not
+/// passed on to the child.
+///
+/// Applying padding to anything but the most mundane sliver is likely to have
+/// undesired effects. For example, wrapping a
+/// [RenderSliverPinnedPersistentHeader] will cause the app bar to overlap
+/// earlier slivers (contrary to the normal behavior of pinned app bars), and
+/// while the app bar is pinned, the padding will scroll away.
+class RenderSliverPadding extends RenderSliverEdgeInsetsPadding {
+  /// Creates a render object that insets its child in a viewport.
+  ///
+  /// The [padding] argument must not be null and must have non-negative insets.
+  RenderSliverPadding({
+    @required EdgeInsetsGeometry padding,
+    TextDirection textDirection,
+    RenderSliver child,
+  }) : assert(padding != null),
+       assert(padding.isNonNegative),
+       _padding = padding,
+       _textDirection = textDirection {
+    this.child = child;
+  }
+
+  @override
+  EdgeInsets resolvedPadding;
```

Comment:
Can you turn this into a private variable with a public getter to ensure that the value cannot be changed from the outside?